src/extensionSlot.ts

```diff
@@ -29,16 +32,21 @@ export function createExtensionSlot<T>(key: SlotKey<T>, host: AppHost, declaring
         getItems,
         getSingleItem,
         getItemByName,
-        discardBy
+        discardBy,
+        subscribe
```

Comment:
I think that while this is very useful for re-rendering slot, it could easily be abused if exposed publicly.
Let's think of a way to make it private and only used by slot renderer.

Reply:
Open to ideas on how to achieve that :)
What kind of "abuse" are you worried about though? I think it might be useful as a public API actually, as slots are not exclusively for react components.

Reply:
@tomenden 
Today we take care of cleanup of contributions, e.g.
```tsx
slot.contribute(fromShell, contribution)
```
If `fromShell` is ejected - the contribution will be ejected as well.


So if you have an API that lets you subscribe to its slot, let's say:
```tsx
createAPI = shell => {
  const slot = shell.declareSlot(slotKey)

  return {
    // ...
    contribute = (fromShell, contribution) => slot.contribute(fromShell, contribution),
    onItemAdded = f => slot.subscribe(f)
  }
}
```
Then, when you subscribe to it, no one will clean up if the subscriber is ejected.

We can privatize the API in a dirty way by just not exporting the full type (as we do with `Shell`/`PrivateShell`).
Or - use the host to interact "secretly" between the slot and `SlotRenderer`
Or - enforce passing shell as identity to keep track of who subscribed so we'll know to clean up

Let me know what you think

Reply:
Take a look at the latest commit and LMK if this is what you meant